src/pages/index.js

```diff
@@ -83,30 +166,55 @@ function createCard(data) {
 
 function handleAddCardFormSubmit(values) {
   const name = values.title;
+  // console.log(values);
   const link = values.URL;
   // generateCard({ name, link }, cardsWrap);
-  const cardEl = createCard({ name, link });
-  cardSection.addItem(cardEl);
+  api.addCards({ title: values.title, url: values.URL }).then((res) => {
+    newCardPopup.setButtonText(true, "Saving...");
+
+    // res is the card data
+    const cardEl = createCard(res);
+    cardSection.addItem(cardEl);
 
-  newCardPopup.close();
+    newCardPopup.close();
+  });
 }
 
 function handleProfileFormSubmit(values) {
-  userInfo.setUserInfo(values);
-  editProfilePopup.close();
+  api.updateUserInfo(values).then((res) => {
+    editProfilePopup.setButtonText(true, "Saving...");
+
+    userInfo.setUserInfo(res);
+    editProfilePopup.close();
+  });
 }
+function handleAvatarFormSubmit(values) {
+  api.updateAvatar(values).then((res) => {
+    editAvatarPopup.setButtonText(true, "Saving...");
+
+    userInfo.setAvatar(res.avatar);
 
+    editAvatarFormValidator.resetValidation();
```

Comment:
`.resetValidation()` is needed only when you open the modal. Please, don't duplicate the call here